docs/apis/subsystems/check/index.md

```diff
@@ -225,6 +225,20 @@ https://github.com/moodle/moodle/blob/master/lib/classes/check/access/riskxss_re
 
 Some checks are by their nature asynchronous. For instance having moodle send an email to itself and then having it processed by the inbound mail handler to make it's properly configured (see [MDL-48800](https://tracker.moodle.org/browse/MDL-48800)). In cases like these please make sure the age or staleness of the check is shown in the summary, and you should also consider turning the result status into a warning if the result is too old. If appropriate make the threshold a configurable admin setting.
 
+## Reusing checks inside admin setting pages
+
+Quite often you have a plugin that includes some sort of status check in the admin settings page where you configure the plugin. Eg you can add the api keys for a service and it shows you right in the settings page if the connection settings work.
+
+You can re-use the Check api to get this nice feature for free:
+
+```php
+$temp->add(new admin_setting_check('antivirus/checkantivirus', new \core\check\environment\antivirus()));
```

Comment:
I'd recommend wrapping this prematurely. I know the coding style is very liberal, but when viewed as code in docs it can lead to horizontal scrollbars being added.

You can see that here: https://deploy-preview-939--moodledevdocs.netlify.app/docs/apis/subsystems/check#reusing-checks-inside-admin-setting-pages if the window isn't obscenely large.

```suggestion
$temp->add(new admin_setting_check(
    'antivirus/checkantivirus',
    new \core\check\environment\antivirus(),
));
```